### PR TITLE
Branches Vi && Steph 

### PR DESCRIPTION
**Pull request by Steph0088**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? | The signature and the block of code inside. It can have parameters but doesn't need them. It is a way to automate code and reduce repetition. 
What are the advantages of using git when collaboratively working on one code base? | You can save changes to different versions of a file and refer back to them as needed. If you make mistakes or want to refer back to a working piece of code, you can find them in git.
What kind of relationship did you and your pair have with the unit tests? | We struggled and were not running them for the first part of our code. It wasn't until the last two waves that we realized that we should have been utilizing them. They actually turned out to be really helpful. The waves were checking conditions that we may have not thought of while testing our own code.
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? | Yes, we used:  each_with_index, slice, include?, split. We used them in our .each iterations, and they helped us sort and move variables from one array to a hash or vice versa. 
What was one method you and your pair used to debug code? | We often used puts statements, as well as finding the lines/error types referred to in error messages when we ran the program. We ended up using the test code in the end to look through our code and find our errors and fails.
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | We would check in and let each other know when we felt lost or when we needed to slow down. We celebrated our success.

### CODE REVIEW

**Comment by jmaddox19** [comment]
# Adagrams
## What We're Looking For


Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | X
Small commits with meaningful commit messages | Some of your commit messages could've been more detailed (like "Made changes for wave 5") and commits could've been made more frequently with smaller amounts of changes, but it looks like y'all are aware of that. It'll come more naturally over time.
**Code Requirements** |
`draw_letters` method | X
Uses appropriate data structure to store the letter distribution | X
All tests for `draw_letters` pass | X
`uses_available_letters?` method | X
All tests for `uses_available_letters?` pass | X
`score_word` method | X
Uses appropriate data structure to store the letter scores | Y'all didn't use a data structure but I'd say your case statement works just as well!
All tests for `score_word` pass | X
`highest_score_from` method | X
Appropriately handles edge cases for tie-breaking logic | X
All tests for `highest_score_from` pass | X
**Overall** | Great job!!!


Y'all did such a good job!! Yay! 👏 👏 👏 


I'm glad y'all found using binding.pry helpful! Also so glad y'all tried out starting on paper! I wish I'd thought to try that out earlier on in my practice with software, would've saved me so much frustration. And so glad y'all were able to learn shortcuts and stuff from each other!

**Comment by tildeee** [comment]
# Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x, instead of having commit messages that say "wave 2 complete" we would rather have commit messages that said something like, "implements the score_word method" or "refactors the letter data structure" etc
**Code Requirements** |
`draw_letters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `draw_letters` pass | x
`uses_available_letters?` method | x
All tests for `uses_available_letters?` pass | x
`score_word` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `score_word` pass | x
`highest_score_from` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highest_score_from` pass | x
**Overall** |

Fantastic work on this project, you two! Your submission has really excellent and readable code. You two made really good decisions about how to keep the code clean and logical. There are even small touches like how you had the comments in the file to describe the methods that were really good.

I have a few comments to point out places where your code could've been cleaned up a little-- this mostly is in some places where "dead code" remained.

That being said, great work overall! :)

File: lib/adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    LETTER_POOL.each do |letter, amount|
+        letters = []
+        i = 0
+        while (i < amount) do
+            letters << letter
+            i += 1
+        end
+        all_letters << letters
+    end
+    all_letters = all_letters.flatten
```
Comment (tildeee) [inline lib/adagrams.rb:21]: NICE! This ends up being so clean and clever!

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+# Wave 3
+
+def score_word(word)
+    
+    letter_points_array = [
+        [],['A', 'E', 'I', 'O', 'U', 'L', 'N','R', 'S', 'T'],
+        ['D','G'],['B', 'C', 'M','P'],['F','H', 'V', 'W', 'Y'],
+        ['K'],[],[],['J', 'X'],[],['Q', 'Z']
+    ]
```
Comment (tildeee) [inline lib/adagrams.rb:63]: Clever data structure!

File: lib/adagrams.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    
+    #This is where we automated our array and turned it into a hash.
+    
+    letter_points_array.each_with_index do |letters, point|
+        letters.each do |letter|
+            letter_points[letter] = point
+        end
+    end
+    puts letter_points
+    puts word.split("")
```
Comment (tildeee) [inline lib/adagrams.rb:75]: Don't forget to delete your `puts` statements before you submit!

File: lib/adagrams.rb [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        score += 8
+    end
+    
+    return score
+    
+end 
+
+# Wave 4
+
+def highest_score_from(words)
```
Comment (tildeee) [inline lib/adagrams.rb:94]: Here, you have a first definition of a method named `highest_score_from`, and then below, you redefine a method with the same name. Ruby ends up using the later definition (because that's what gets defined last), but Ruby still gives a warning `warning: method redefined; discarding old highest_score_from`. Feel free to delete this method if it doesn't get used in the end!

File: lib/adagrams.rb [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    end
+    
+    return {word: top_word, score: top_score}
+end
+
+def highest_score_from(words)
+    top_word = ""
+    top_score = 0
+    tie_breaker_word = ""
+    tie_breaker_score = ""
```
Comment (tildeee) [inline lib/adagrams.rb:120]: The variable `tie_breaker_score` ends up not being used anywhere else in the file-- is it safe to delete?
